src/global/Components/QrScanner/QrScanner.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useCallback } from 'react';
+import styled from 'styled-components';
+import Webcam from 'react-webcam';
+import { QRCode } from 'jsqr';
+import getPixels from 'get-pixels';
+/* eslint-disable */
+//@ts-ignore
+import Worker from 'worker-loader!./decodeWorker.worker';
+import VideoOverlay from './VideoOverlay';
+
+const Wrapper = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
+const Button = styled.button`
+  color: white;
+  opacity: 0.5;
+  width: 50px;
+  height: 50px;
+  border-radius: 50%;
+  display: flex;
+  position: absolute;
+  bottom: 30px;
+  left: 50%;
+  transform: translate(-50%);
+`;
+
+type ExtendedWebcam = {
+  ref: React.RefObject<Webcam>;
+};
+
+const StyledWebcam = styled(Webcam)<ExtendedWebcam>`
+  width: 100%;
+  height: 100vh;
+  object-fit: fill;
+`;
+
+type defaultProps = {
+  videoConstraints: MediaTrackConstraints;
+  interval: number;
+  buttonMode: boolean;
+  className: string;
+  onError: (error: string) => void;
+};
+
+type Props = {
+  onCode: (code: QRCode) => void;
```

Comment:
maybe this methods can be optional so we can have one type QrScannerProps ?

Reply:
It doesn't make too much sense to be optional, as we wouldn't use scanning then